elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -76,6 +72,12 @@
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
+import static com.yahoo.elide.annotation.LifeCycleHookBinding.Operation.CREATE;
```

Comment:
I already fixed all CheckStyle errors today with this [commit](https://github.com/yahoo/elide/pull/2283/commits/ab241565a2714b14ee040349135d00d68b72288b)

Do let me know if there is still any CheckStyle error.

Reply:
No checkstyle errors, but other IT tests are failing because of the change to the data model.  Check the screwdriver build:

https://cd.screwdriver.cd/pipelines/6103/builds/733421/steps/build

I recommend building all of elide locally.

Reply:
1. Fixed a few failed tests caused by the changes. Also, found and fixed a corner case with [this commit](https://github.com/yahoo/elide/pull/2283/commits/9bff2c60f7d41d16ceff8b550800b6cfae7fc5b6)

2. Tried building all of elide locally and found a few existing but failed tests (both unit and IT tests) that are not caused by the changes. Fixed one of them like [this](https://github.com/yahoo/elide/pull/2283/commits/4859765ac90f3ad1bbfe4a0995c49dcc67e77842)
but want to double check if you are aware of them, for example [this one](https://github.com/yahoo/elide/blob/master/elide-integration-tests/src/test/java/com/yahoo/elide/tests/UserTypeIT.java)
This surprised me by the way.

3. We want to rebase this onto the elide-6.x but my fork does not have that branch yet. To minimize work, I want to recreate a fork, rebase and then open a new PR for the changes. Your thoughts?

Reply:
> 1. Fixed a few failed tests caused by the changes. Also, found and fixed a corner case with [this commit](https://github.com/yahoo/elide/pull/2283/commits/9bff2c60f7d41d16ceff8b550800b6cfae7fc5b6)

When is newVal not a map? 

> 2. Tried building all of elide locally and found a few existing but failed tests (both unit and IT tests) that are not caused by the changes. Fixed one of them like [this](https://github.com/yahoo/elide/pull/2283/commits/4859765ac90f3ad1bbfe4a0995c49dcc67e77842)

Makes sense.

>    but want to double check if you are aware of them, for example [this one](https://github.com/yahoo/elide/blob/master/elide-integration-tests/src/test/java/com/yahoo/elide/tests/UserTypeIT.java)

I would need to see the test failure and why it is failing (unfortunately the screwdriver build is not building the branch and so it never gets here).

>    This surprised me by the way.
> 3. We want to rebase this onto the elide-6.x but my fork does not have that branch yet. To minimize work, I want to recreate a fork, rebase and then open a new PR for the changes. Your thoughts?

I think that sounds like a good plan.

Reply:
That's because this particular model has an attribute of type Object.

private Object obj = "foo";

which is really of type String.  Elide thinks it is a complex attribute because of its type.

Reply:
Here is the GraphQL query:

https://github.com/yahoo/elide/blob/master/elide-graphql/src/test/resources/graphql/responses/upsert/rootSingleWithId.json

So yes, it appears it is possible to be a non-map.